lib/OpenQA/Worker/Cache/Task/Asset.pm

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Task::Asset;
+
+use Mojo::Base 'OpenQA::Worker::Cache::Task';
+use Mojo::URL;
+use constant LOCK_RETRY_DELAY   => 30;
+use constant MINION_LOCK_EXPIRE => 99999;    # ~27 hours
+
+use OpenQA::Worker::Cache::Client;
+use OpenQA::Worker::Cache;
+use OpenQA::Worker::Cache::Request;
+
+has cache => sub { OpenQA::Worker::Cache->from_worker };
+
+sub register {
+    my ($self, $app) = @_;
+
+    $app->minion->add_task(
+        cache_asset => sub {
+            my $job = shift;
+            my ($id, $type, $asset_name, $host) = @_;
+            my $req = OpenQA::Worker::Cache::Request->new->asset(
+                id    => $id,
+                type  => $type,
+                asset => $asset_name,
+                host  => $host
+            );
+            my $guard_name = $self->_gen_guard_name($req->lock);
+
+            return $job->remove unless defined $asset_name && defined $type && defined $host;
+            return $job->retry({delay => LOCK_RETRY_DELAY})
+              unless my $guard = $app->minion->guard($guard_name, MINION_LOCK_EXPIRE);
+
+            my $job_prefix = "[Job #" . $job->id . "]";
+            $app->log->debug("${job_prefix} Guard: $guard_name Download: $asset_name");
```

Comment:
Looks like `${job_prefix}` is a leftover from a previous version and should be `$job_prefix`?

Reply:
Done in https://github.com/os-autoinst/openQA/pull/1783/commits/71e142c959efcf6c2e395f891e7522869be78d7d